pkg/correlation/hook.go

```diff
@@ -0,0 +1,157 @@
+package correlation
```

Comment:
missing APACHE 2 LIC header in all new files. Need to be added.

Reply:
adding